### PR TITLE
fix Model docstring

### LINKED ISSUE #6040: Model docstring is outdated
We no longer have self.Var("name", dist) to register random variables. We need to fix that with a correct API call

https://github.com/pymc-devs/pymc/blob/ad16bf48a1703b80cee3d3fef9df1dd0ae552d7c/pymc/model.py#L469

### PR DESCRIPTION
**Pull request by alekracicot**
**What is this PR about?**
...
Description: Change Model docstring 
Issue name Model docstring is outdated
Closes  #6040

**Checklist**
+ [x ] Explain important implementation details 👆
+ [x ] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html). (really just changed the docstring.
+ [ x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x ] Are the changes covered by tests and docstrings? No test needed, only doc)
+ [ x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Fix docstring for the Model class, replacing var by register_rv

### CODE REVIEW

**Comment by michaelosthege** [comment]
There are a few mentions of the `pm.Model` class that don't have a `pm.` in front yet.

I tried to commit to your branch to get this over the finish line, but somehow I couldn't. (At least not in GitHub.)

@alekracicot can you insert the `pm.` also for the mentions of `Model` in the docstring?

**Comment by alekracicot** [comment]
Yeah absolutely, I'll do that right now.

**Comment by alekracicot** [comment]
@michaelosthege, I added the changes.

**Review by twiecki — APPROVED** [review]

**Comment by reshamas** [comment]
Hi @alekracicot 
Did you learn of contributing to PyMC from this website?
https://pymc-data-umbrella.xyz/en/latest/2022-07_sprint/schedule.html

Or some other avenue?

**Comment by alekracicot** [comment]
@reshamas No mostly the contributing section on the pymc website.

File: pymc/model.py [hunk 1]
```diff
@@ -466,51 +466,50 @@ def __init__(self, mean=0, sigma=1, name=''):
                 # will get model's name prefix
 
                 # 3) you can create variables with Var method
```
Comment (ricardoV94) [inline pymc/model.py]: This still mentions `var` method

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

File: pymc/model.py [hunk 2]
```diff
@@ -466,51 +466,50 @@ def __init__(self, mean=0, sigma=1, name=''):
...
-                self.Var('v1', Normal.dist(mu=mean, sigma=sd))
-                # this will create variable named like '{prefix::}v1'
+                self.register_rv(pm.Normal.dist(mu=mean, sigma=sigma), 'v1', initval=1)
```
Comment (ricardoV94) [inline pymc/model.py]:
```suggestion
                self.register_rv(Normal.dist(mu=mean, sigma=sigma), 'v1', initval=1)
```

Why are we adding the `pm` prefix and showing `initval`?

File: pymc/model.py [hunk 3]
```diff
@@ -465,52 +465,51 @@ def __init__(self, mean=0, sigma=1, name=''):
                 # variables in several ways note, that all variables
                 # will get model's name prefix
 
-                # 3) you can create variables with Var method
-                self.Var('v1', Normal.dist(mu=mean, sigma=sd))
-                # this will create variable named like '{prefix::}v1'
+                # 3) you can create variables with the register_rv method
+                self.register_rv(Normal.dist(mu=mean, sigma=sigma), 'v1', initval=1)
```
Comment (ricardoV94) [inline pymc/model.py:469]: was the initval needed?

**Review by ricardoV94 — COMMENTED** [review]
there are still new `pm.` bring introduced in the rest of the changes

File: pymc/model.py (hunk 3 above)
Comment (alekracicot) [reply pymc/model.py:469]: I don't think its needed no, but it was a recommended change so I kept it there.

Comment (ricardoV94) [reply pymc/model.py:469]: There are other more important arguments that can be passed to `register_rv`. I don't know what was the rationale to show this one in particular.

Comment (ricardoV94) [reply pymc/model.py:469]: But we can keep it as another example below shows the initval as well

Comment (alekracicot) [reply pymc/model.py:469]: Allright sounds good

**Comment by alekracicot** [comment]
> 

Ok, I will remove the pm.

File: pymc/model.py [hunk 4]
```diff
@@ -465,18 +465,18 @@ def __init__(self, mean=0, sigma=1, name=''):
...
                 # variable can be accessed with self.v1 or self['v1']
 
                 # 4) this syntax will also work as we are in the
                 # context of instance itself, names are given as usual
-                Normal('v2', mu=mean, sigma=sd)
+                Normal('v2', mu=mean, sigma=sigma)
 
                 # something more complex is allowed, too
-                half_cauchy = HalfCauchy('sigma', beta=10, initval=1.)
+                half_cauchy =  HalfCauchy('sd', beta=10, initval=1.)
```
Comment (ricardoV94) [inline pymc/model.py]: Reverting unrelated changes

```suggestion
                half_cauchy = HalfCauchy('sigma', beta=10, initval=1.)
```

File: pymc/model.py [hunk 5]
```diff
@@ -510,7 +510,6 @@ def __init__(self, mean=0, sigma=1, name=''):
             CustomModel(mean=2, name='second')
 
         # variables inside both scopes will be named like `first::*`, `second::*`
-
```
Comment (ricardoV94) [inline pymc/model.py:513]: Removed line by accident?

Comment (alekracicot) [reply pymc/model.py:513]: No, the rest of the code does not introduce an extra blank line at the end of the docstring.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks @alekracicot!
